### PR TITLE
support 32-bit ARM (aarch32) syscalls

### PR DESCRIPTION
**Pull request by kkourt**
This PR adds support for dealing with 32-bit ARM syscalls on arm64.
See patches.

```release-note
tracing: support 32-bit ARM (aarch32) syscalls
```

### CODE REVIEW

**Comment by kkourt** [comment]
Checkpatch failure is on existing code:
```
  CHECK:CAMELCASE: Avoid CamelCase: <is32BitSyscall>
  #50: FILE: bpf/process/generic_calls.h:155:
  +		e->sel.is32BitSyscall = flags & (1 << TIF_32BIT);
  ```

**Review by mtardy — APPROVED** [review]
looks good, to be fair the first commit is tricky to review, the list code is a bit hard to read. Do we have tests that cover policies with list usage?

File: pkg/sensors/tracing/lists.go [hunk 1]
```diff
@@ -7,22 +7,26 @@ import (
...
 	"github.com/cilium/tetragon/pkg/btf"
 	"github.com/cilium/tetragon/pkg/ftrace"
 	gt "github.com/cilium/tetragon/pkg/generictypes"
 	"github.com/cilium/tetragon/pkg/k8s/apis/cilium.io/v1alpha1"
-	"github.com/cilium/tetragon/pkg/syscallinfo"
 )
 
-func getList(name string, lists []v1alpha1.ListSpec) *v1alpha1.ListSpec {
+// isList checks if a value specifics a list, and if so it returns it (or nil if list does not exist)
+func isList(val string, lists []v1alpha1.ListSpec) (bool, *v1alpha1.ListSpec) {
```
Comment (olsajiri) [inline pkg/sensors/tracing/lists.go:17]: do we need to return bool? returning just list and callers checking it for nil should be enought, right?

Comment (olsajiri) [reply pkg/sensors/tracing/lists.go:17]: nah it's used in generic functions like getKprobeSymbols to recognize list, ok then ;-)

**Review by olsajiri — APPROVED** [review]

File: pkg/sensors/tracing/syscall_list.go [hunk 2]
```diff
@@ -0,0 +1,141 @@
+// SPDX-License-Identifier: Apache-2.0
```
Comment (olsajiri) [inline pkg/sensors/tracing/syscall_list.go:1]: I like that, code is more structured now, thanks

File: pkg/observer/observertesthelper/observer_test_helper.go [hunk 3]
```diff
@@ -423,6 +423,9 @@ func loadObserver(tb testing.TB, ctx context.Context, base *sensors.Sensor,
 	if err := base.Load(option.Config.BpfDir); err != nil {
 		tb.Fatalf("Load base error: %s\n", err)
 	}
+	tb.Cleanup(func() {
+		base.Unload()
+	})
```
Comment (willfindlay) [inline pkg/observer/observertesthelper/observer_test_helper.go:428]: Amazing fix, thank you

File: pkg/sensors/tracing/syscall_list.go [hunk 4]
```diff
@@ -51,6 +51,8 @@ func (v SyscallVal) Symbol() (string, error) {
 		prefix = "__arm64_"
 	case "i386":
 		prefix = "__ia32_"
+	case "arm32":
+		prefix = "__arm64_"
```
Comment (willfindlay) [inline pkg/sensors/tracing/syscall_list.go:55]: Could we add a comment here explaining the same thing you explained in the commit message about the `__arm64_` prefix?

**Review by willfindlay — APPROVED** [review]
LGTM overall. Just a couple nits, aside from the typos @mtardy already pointed out. Pre-approving now.

File: .github/workflows/gotests.yml [hunk 5]
```diff
@@ -46,6 +40,17 @@ jobs:
         echo `which llc`
         echo `clang --version`
 
+
+    - name: Install dependencies x86
+      run: |
+        sudo apt-get -y install libc6-dev-i386
+      if: ${{ matrix.os == 'ubuntu-20.04' }}
```
[the remark this answers is not in the document]
Comment (kkourt) [reply .github/workflows/gotests.yml:47]: Thanks, leaving it as is for now.

File: docs/content/en/docs/concepts/tracing-policy/hooks.md [hunk 6]
```diff
@@ -535,7 +535,7 @@ spec:
     type: "syscalls"
     values:
     - "sys_dup"
-    - "__ia32_sys_dup"
+    - "i386/sys_dup"
```
[the remark this answers is not in the document]
Comment (kkourt) [reply docs/content/en/docs/concepts/tracing-policy/hooks.md:538]: It's not a breaking change. The old format is still supported, but we are using a new format. So everything that used to work with previous versions, will still work.

File: pkg/sensors/tracing/syscall_list.go (hunk 4 above)
Comment (kkourt) [reply pkg/sensors/tracing/syscall_list.go:55]: God point, added a comment.

**Comment by kkourt** [comment]
Thanks for the reviews!
I believe all the feedback is addressed, merging.

**Pull request merged**